Expense/expenseRouter.js

```diff
@@ -0,0 +1,48 @@
+const express = require('express')
+const router = express.Router()
+const Expense = require('./expense.js')
+
+const handleError = (err, req, res, next) => {
+    if (err.errors["amount"]) {
+        return res.status(400).json({ errorMessage: err.errors["amount"].message })
+    } else if (err.errors["description"]) {
+        return res.status(400).json({ errorMessage: err.errors["description"].message })
+    } else if (err.errors["budget"]) {
+        return res.status(400).json({ errorMessage: "Invalid object id for Budget." })
+    } else if (err.errors["category"]) {
+        return res.status(400).json({ errorMessage: "Invalid object id for category." })
+    } else {
+        return res.status(500).json({ errorMessage: "There was an error while saving budget to the database." })
+    }
+}
+
+router.route("/")
```

Comment:
Instead of sending the `ObjectId` along with the client request, we can just send a POST request that looks like this:

```
{
	"amount": 35,
	"description": "potatoes",
	"budget": "Monthly Spending",
	"category": "Groceries"
}
```

In order to do that, we need to add some logic to our route:

```
.post((req, res, next) => {
  let { amount, budget, category, description } = req.body

  budget = Budget.findOne({ title: budget })
  category = Category.findOne({ title: category })

  Promise.all([budget, category])
    .then(([budget_id, category_id]) => {
      const expense = new Expense({
        amount,
        budget: budget_id,
        category: category_id,
        description
      })

      expense.save()
        .then(response => res.status(201).json(response))
        .catch(err => next(err))
    })
    .catch(err => next(err))
  })
```

We have to get the `ObjectId` of `budget` and `category` with the `findOne` method, then using `Promise.all`, we grab the `ObjectId` from the response from MongoDB, then create our new Expense.